### PR TITLE
Garbage collector e2e tests create deployments with unique labels, and unique podTemplate

### PR DESCRIPTION
**Pull request by caesarxuchao**
Fixes https://github.com/kubernetes/kubernetes/issues/49427

The bug was because two tests point to the same `podTemplate`. One [test](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apimachinery/garbage_collector.go#L583) changing the "labels" caused the deployment creation in the other test failing, because the selector doesn't match the labels anymore.

### CODE REVIEW

**Comment by caesarxuchao** [comment]
There are more affected tests. I'm keeping it to myself for now.

**Comment by caesarxuchao** [comment]
Updated.

File: test/e2e/apimachinery/garbage_collector.go [hunk 1]
```diff
@@ -571,16 +586,18 @@ var _ = SIGDescribe("Garbage collector", func() {
...
 		if err != nil {
 			framework.Failf("Failed to create replication controller: %v", err)
 		}
 		rc2Name := "simpletest-rc-to-stay"
-		rc2 := newOwnerRC(f, rc2Name, 0)
-		rc2.Spec.Selector = nil
-		rc2.Spec.Template.ObjectMeta.Labels = map[string]string{"another.key": "another.value"}
+		// TODO: find better way to keep this label unique in the test
+		uniqLabels = map[string]string{"another.key": "another.value"}
+		rc2 := newOwnerRC(f, rc2Name, 0, uniqLabels)
```
Comment (MrHohn) [inline test/e2e/apimachinery/garbage_collector.go]: We used to set `rc2.Spec.Selector = nil`, is that still needed?

Comment (caesarxuchao) [inline test/e2e/apimachinery/garbage_collector.go]: No, nil selector will be default to match the podTemplate labels by the server.

**Review by MrHohn — APPROVED** [review]

**Comment by caesarxuchao** [comment]
Thanks @MrHohn

**Review by janetkuo — APPROVED** [review]

**Pull request merged**